src/rust/scheduler/scheduler.rs

```diff
@@ -257,12 +292,38 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_scheduler_insert() -> Result<()> {
+        let scheduler: Scheduler = Scheduler::default();
+
+        // Insert a task and make sure the task id is not a simple counter.
+        let task: DummyTask = DummyTask::new(String::from("testing"), Box::pin(DummyCoroutine::new(0)));
+        let handle: SchedulerHandle = match scheduler.insert(task) {
+            Some(handle) => handle,
+            None => anyhow::bail!("insert() failed"),
+        };
+        let task_id: u64 = handle.into_raw();
+        crate::ensure_neq!(task_id, 0);
```

Comment:
The pseudo-random may return you `0` and this assert would fail. We should remove it.